spec/models/rook_spec.rb

```diff
@@ -0,0 +1,54 @@
+require 'rails_helper'
+
+RSpec.describe Rook, type: :model do
+  describe "#valid_move?" do
+    let(:game) {create :game}
+
+    it "should properly evaluate malformed moves that are otherwise valid" do
+      white_rook = game.pieces.find_by(rank: 1, file: 1)
+      black_rook = game.pieces.find_by(rank: 8, file: 1)
+      expect(white_rook.valid_move?(0, 1)).to eq(false)
+      expect(white_rook.valid_move?(1, 0)).to eq(false)
+      expect(white_rook.valid_move?(0, 0)).to eq(false)
+      expect(white_rook.valid_move?(0, 2)).to eq(false)
+      expect(white_rook.valid_move?(2, 0)).to eq(false)
+      expect(white_rook.valid_move?(1, 1)).to eq(false)
+      expect(black_rook.valid_move?(8, 1)).to eq(false)
+      expect(black_rook.valid_move?(7, 0)).to eq(false)
+      expect(black_rook.valid_move?(8, 0)).to eq(false)
+      expect(black_rook.valid_move?(9, 0)).to eq(false)
+      expect(black_rook.valid_move?(9, 1)).to eq(false)
+      expect(black_rook.valid_move?(9, 2)).to eq(false)
+    end
```

Comment:
Could you please cover the following cases here as well?

- A rank that is too large and a file that is too large
- A rank that is too small and a file that is too large
- A valid rank and a file that is too large